WordPress/src/main/java/org/wordpress/android/ui/jetpack/scan/ScanFragment.kt

```diff
@@ -3,26 +3,68 @@ package org.wordpress.android.ui.jetpack.scan
 import android.os.Bundle
 import android.view.View
 import androidx.fragment.app.Fragment
+import androidx.lifecycle.Observer
 import androidx.lifecycle.ViewModelProvider
+import kotlinx.android.synthetic.main.scan_fragment.recycler_view
 import org.wordpress.android.R
 import org.wordpress.android.WordPress
+import org.wordpress.android.fluxc.model.SiteModel
+import org.wordpress.android.ui.jetpack.scan.ScanViewModel.UiState.Content
+import org.wordpress.android.ui.jetpack.scan.adapters.ScanAdapter
+import org.wordpress.android.ui.utils.UiHelpers
 import javax.inject.Inject
 
 class ScanFragment : Fragment(R.layout.scan_fragment) {
+    @Inject lateinit var uiHelpers: UiHelpers
     @Inject lateinit var viewModelFactory: ViewModelProvider.Factory
     private lateinit var viewModel: ScanViewModel
 
-    override fun onCreate(savedInstanceState: Bundle?) {
-        super.onCreate(savedInstanceState)
-        (requireActivity().application as WordPress).component().inject(this)
-    }
-
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
-        initViewModel()
+
+        val nonNullActivity = requireActivity()
+        (nonNullActivity.application as WordPress).component()?.inject(this)
+
+        val site = if (savedInstanceState == null) {
```

Comment:
I think we can update FluxC code to just use `siteId`. We can eliminate this passing and retrieval of the entire site model. Lmkwyt.

Reply:
👋 @ashiagr !

I think you are right and I agree. Since only `siteId` is the only property needed and considering that no other property from the entire site model will be used in the future, I believe the best course of action is to replace `site` with `siteId`.

IMHO, the main gain from doing something like that is to make it explicit to the reader that this is the only thing needed, so they know straight away and can score the feature down a bit. Obviously, there is a performance optimization as well, but to be honest this will not save us much.

Reply:
I wanted a second opinion if we should be doing it, thanks for sharing your thoughts 🙏. I'll not make any changes as of now, may be just add "nice to have" in the backlog.

Reply:
Yeap, definitely! But, on the other hand, it might be decided and for a good reason to keep using whole models instead, confronting my argument above, so I keep my mind open, as it indeed can help not having to make two PRs every time you want to add an extra property from the model or the whole model next time you needs more from it. 🤔